local-volume/provisioner/pkg/util/volume_util.go

```diff
@@ -186,15 +197,31 @@ func (u *FakeVolumeUtil) IsDir(fullPath string) (bool, error) {
 
 	for _, f := range files {
 		if file == f.Name {
-			return !f.IsNotDir, nil
+			if f.VolumeType != FakeEntryFile {
+				// Accurately simulate how a check on a non file returns error with actual OS call.
+				return false, fmt.Errorf("%q not a file or directory", fullPath)
+			}
+			return true, nil
 		}
 	}
-	return false, fmt.Errorf("File %q not found", fullPath)
+	return false, fmt.Errorf("Directory entry %q not found", fullPath)
 }
 
 // IsBlock checks if the given path is a block device
 func (u *FakeVolumeUtil) IsBlock(fullPath string) (bool, error) {
-	return false, nil
+	dir, file := filepath.Split(fullPath)
+	dir = filepath.Clean(dir)
+	files, found := u.directoryFiles[dir]
+	if !found {
+		return false, fmt.Errorf("Directory %q not found", dir)
+	}
+
+	for _, f := range files {
+		if file == f.Name {
+			return f.VolumeType == FakeEntryBlock, nil
```

Comment:
@msau42 : The reason that I return an err in isDir() is because I found the real code exiting with an err when the path was not a dir or file (because the OS responds with an error when open is called on a non dir/file). I decided to simulate this accurately in the mock (which it was not doing earlier) and hence an error is returned. This behavior is not seen by isBlock() and hence does not need to be mocked.